module/VuFind/src/VuFind/GeoFeatures/BasemapConfig.php

```diff
@@ -38,22 +36,22 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org/wiki/development:plugins:hierarchy_components Wiki
  */
-class BasemapConfig
+class BasemapConfig extends AbstractConfig
 {
     /**
      * Basemap tileserver URL
      * Default is the wikimedia osm-intl map
      *
      * @var string
      */
-    protected $basemapUrl = "https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}.png";
+    protected $basemap_url = "https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}.png";
```

Comment:
Do these basemap_url and basemap_attribution properties serve any purpose? It doesn't look like they're being read anywhere. Also note that for style reasons, if we do need these, it's probably preferable to keep the old camelCase names instead of the under_scored versions... but I'm guessing that we may just be able to remove them entirely.

Reply:
Oh, wait, I see what's happening -- these are getting dynamically loaded as defaults. I guess I can live with the less-than-ideal naming as a matter of necessity for consistency with the .ini files. The only alternative would be to have a $defaultOptions property that holds all of these values, instead of having a separate property for each configuration.

Reply:
I suppose yet another possibility would be to have a getDefaultOptions function, so that instead of doing the for loop to populate the options array from class properties, you just have:

<pre>
if (empty($options)) {
    return $this->getDefaultOptions();
}
</pre>

and then elsewhere define:

<pre>
protected function getDefaultOptions()
{
    return [
        'option1' => 'value1',
        'option2' => 'value2';
    ];
}
</pre>

The difference between a defaultOptions property and a getDefaultOptions() function is pretty minimal in most scenarios, but it has some impact on how your code formatting looks (I feel like with a lot of options, the function might look cleaner and make commenting easier) and also how extensibility works (somebody subclassing the code would need to override defaultOptions property details in a custom constructor... while a getDefaultOptions function could be overridden directly).

I don't honestly have a strong opinion about this -- just laying out the options. Sorry if I'm wasting too much time on a trivial matter. :-)

Reply:
@demiankatz, 
What if we change the .ini file configuration to use camelCase for the configuration parameters (so we use basemapUrl and basemapAttribution)? It would make the code so that it isn't backwards compatible, but given we are doing a major upgrade, I wonder if that would be okay. What do you think?

Also, I'll take a look into implementing a getDefaultOptions() function tomorrow.

Reply:
If memory serves, the basemap settings are all new anyway, so changing them won't cause an incompatibility with a previous VuFind release... it will only affect people who are working on bleeding-edge master, of which few if any are using this new geographic feature. In other words, unless my memory is playing tricks, I think it's safe to change this... but you may find that it's unnecessary if getDefaultOptions proves a cleaner solution anyway.